app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -376,12 +377,15 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
-    fun onSaveFile(assetName: String?, assetData: ByteArray) {
+    fun onSaveFile(assetName: String?, assetData: ByteArray, messageId: String) {
         viewModelScope.launch {
             withContext(dispatchers.io()) {
                 fileManager.saveToExternalStorage(assetName, assetData) {
-                    onFileSavedToExternalStorage(assetName)
-                    hideOnAssetDownloadedDialog()
+                    viewModelScope.launch {
+                        updateAssetMessageDownloadStatus(SAVED_EXTERNALLY, conversationId, messageId)
+                        onFileSavedToExternalStorage(assetName)
+                        hideOnAssetDownloadedDialog()
```

Comment:
Instead of launching 2 coroutines, I would suggest to make `saveToExternalStorage` and `onFileSaved` suspend and change there the the dispatcher

    suspend fun saveToExternalStorage(assetName: String?, assetData: ByteArray,  onFileSaved: suspend (String?) -> Unit) {
        withContext(dispatcher.IO) {
            saveFileToDownloadsFolder(assetName, assetData, context)
            onFileSaved(assetName)
        }
    }

This way the calling suspend functions don't have to worry about what Dispatcher to use. This responsibility lies in the class that does the work.